src/interfaces.ts

```diff
@@ -59,7 +59,33 @@ export enum AuthenticationTypes {
 
 export type AuthenticationType = `${AuthenticationTypes}`;
 
-export type AuthenticationOptions = { apiKeyHeaderName?: string };
+interface CommonSecurityDefinition {
+  id: string;
+}
+
+export interface BasicSecurityDefinition extends CommonSecurityDefinition {
+  type: AuthenticationTypes.BASIC;
+}
+
+export interface ApiKeySecurityDefinition extends CommonSecurityDefinition {
+  type: AuthenticationTypes.API_KEY;
+  header: string;
+}
+
+export type SecurityDefinition =
+  | BasicSecurityDefinition
+  | ApiKeySecurityDefinition;
+
+/**
+ * Simplified security object.
+ * @example [['MyId'], ['MyId2', 'MyId3]] becomes [{ MyId: [] }, { MyId2: [], MyId3: [] }
+ */
+export type Security = string[][];
```

Comment:
I suppose this could just define the `Security` exactly as swagger expects it for pretty minimal additional boilerplate...!